### DISCUSSION TITLE
Using postgis in javascript project: "column x.ST_DWithin does not exist" error

### DISCUSSION THREAD

**Discussion by Ivaskuu** [supabase/supabase#9424] [Questions]
Hi there! I'm pretty new to supabase and sql, so sorry if I'm missing something.

I am trying to call this sql query on the supabase SQL Editor, which looks for Announcements within a 3km range, which works fine and returns a row.
```
SELECT ST_Distance(
  'SRID=4326;POINT(45.6478009 9.6295856)'::geography,
  "Announcement".possible_location
) as distance, *
FROM "Announcement"
WHERE ST_DWithin('SRID=4326;POINT(45 9)'::geography, "Announcement".possible_location, 3000);
```

So I converted it to a supabase-js query to call from my node-js server like this (I hope it is correct, I've replaced the WHERE clause with .lte()):
```
const { data, error } = await supabase
	.from('Announcement')
	.select(`*`)
	.lte(`ST_DWithin('SRID=4326;POINT(45 9)'::geography, Announcement.possible_location)`, '3000');
```

Unfortunately I get this error:
```
{
    "code": "42703",
    "details": null,
    "hint": null,
    "message": "column Announcement.ST_DWithin does not exist"
}
```

How can I tell supbase that I don't want to search for ST_DWithin property inside Announcement table, but to use the postgis extension?
I don't want to use the .rpc() and create a function, this is a query I'll do a lot of times and I'm afraid of exceeding the max free tier rpc calls.
Thank you, any help is appreciated!

**Comment by GaryAustin1** [comment]
Their is no limit on rpc calls which use a database function, they are treated just like select requests as far a limits.  You might be thinking of edge functions.

Normally you can't run postgres functions from the api with filters.
